sdk/scheduler/src/main/java/com/mesosphere/sdk/api/PodResource.java

```diff
@@ -165,6 +172,107 @@ public Response getPodInfo(@PathParam("name") String name) {
     }
 
     /**
+     * Restarts a pod in a "stopped" debug mode.
+     */
+    @Path("/{name}/stop")
+    @POST
+    public Response stopPod(@PathParam("name") String name, String bodyPayload) {
+        Set<String> taskFilter;
+        try {
+            taskFilter = parseJsonList(bodyPayload);
+        } catch (JSONException e) {
+            LOGGER.error(String.format("Failed to parse task filter '%s'", bodyPayload), e);
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        try {
+            return overrideGoalState(name, taskFilter, GoalStateOverride.STOPPED);
+        } catch (Exception e) {
+            LOGGER.error(String.format("Failed to stop pod '%s' with task filter '%s'", name, taskFilter), e);
+            return Response.serverError().build();
+        }
+    }
+
+    /**
+     * Restarts a pod in a normal state following a prior "stop" command.
+     */
+    @Path("/{name}/start")
+    @POST
+    public Response startPod(@PathParam("name") String name, String bodyPayload) {
+        Set<String> taskFilter;
+        try {
+            taskFilter = parseJsonList(bodyPayload);
+        } catch (JSONException e) {
+            LOGGER.error(String.format("Failed to parse task filter '%s'", bodyPayload), e);
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        try {
+            return overrideGoalState(name, taskFilter, GoalStateOverride.NONE);
+        } catch (Exception e) {
+            LOGGER.error(String.format("Failed to start pod '%s' with task filter '%s'", name, taskFilter), e);
+            return Response.serverError().build();
+        }
+    }
+
+    private static Set<String> parseJsonList(String payload) throws JSONException {
```

Comment:
I think these internal methods should get out of the `PodResource` class and go somewhere else.  I prefer these classes having very little logic in them.

Reply:
Added a `RequestUtils` class with this method as well as a method to filter tasks